ObjectiveGit/GTRepository.m

```diff
@@ -948,4 +951,73 @@ - (BOOL)cleanupStateWithError:(NSError * _Nullable __autoreleasing *)error {
 	return YES;
 }
 
+#pragma mark Notes
+
+- (GTNote *)createNote:(NSString *)note target:(GTObject *)theTarget referenceName:(NSString *)referenceName author:(GTSignature *)author committer:(GTSignature *)committer overwriteIfExists:(BOOL)overwrite error:(NSError **)error {
+	git_oid oid;
+	
+	int gitError = git_note_create(&oid, self.git_repository, referenceName.UTF8String, author.git_signature, committer.git_signature, theTarget.OID.git_oid, [note UTF8String], overwrite ? 1 : 0);
+	if (gitError != GIT_OK) {
+		if (error != NULL) *error = [NSError git_errorFor:gitError description:@"Failed to create a note in repository"];
+		
+		return nil;
+	}
+	
+	return [[GTNote alloc] initWithTargetOID:theTarget.OID repository:self referenceName:referenceName error:error];
```

Comment:
I wasn't clear (also, do realise I have no prior knowledge of how notes actually work, apart from a quick glance at the docs that told me that notes are blobs 😉).

From what I can tell, the `oid` above is the one that is created by `git_note_create`, a.k.a the `oid` of the blob that got added with the note contents as its data. `git_note_read` does a re-lookup "from the start" — ie. grabs the ref, does "something", and populates the `git_note` object we want to wrap.